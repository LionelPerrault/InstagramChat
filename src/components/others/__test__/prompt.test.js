import React from 'react'
import { shallow } from 'enzyme'
import Prompt from '../prompt'
import { create } from 'react-test-renderer'

describe('Prompt Component', () => {
  const mockFn = jest.fn()
  const defaultProps = {
    title: 'Title of the prompt component',
    content: 'Some content here..',
    actionText: 'Delete',
    action: mockFn,
    back: mockFn,
  }

  it('should match snapshot', () => {
    const tree = create(<Prompt {...defaultProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot with blurred prop', () => {
    const tree = create(<Prompt {...defaultProps} blurred />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should simulate click events', () => {
    const wrapper = shallow(<Prompt {...defaultProps} />)

    wrapper.find('PrimaryButton').simulate('click')
    wrapper.find('SecondaryButton').simulate('click', { preventDefault() {} })
  })
})
