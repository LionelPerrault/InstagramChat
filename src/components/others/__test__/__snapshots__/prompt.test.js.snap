// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prompt Component should match snapshot 1`] = `
Array [
  <div
    className="overlay"
    onClick={null}
    style={
      Object {
        "cursor": "inherit",
        "opacity": 0.5,
      }
    }
  />,
  <div
    className="prompt"
  >
    <div
      style={
        Object {
          "animation": "bcCCNc 300ms ease 0s 1 normal none running",
          "backfaceVisibility": "visible",
        }
      }
    >
      <div
        className="prompt-top"
      >
        <span
          className="prompt-title"
        >
          Title of the prompt component
        </span>
        <span
          onClick={[Function]}
        >
          <i
            className="material-icons"
          >
            close
          </i>
        </span>
      </div>
      <div
        className="prompt-middle"
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <span
          className="prompt-content"
        >
          Some content here..
        </span>
      </div>
      <div
        className="prompt-bottom"
      >
        <a
          className="sec_btn"
          href="#"
          onClick={[Function]}
        >
          Cancel
        </a>
        <a
          className="pri_btn prompt-done"
          href="#"
          onClick={[MockFunction]}
        >
          Delete
        </a>
      </div>
    </div>
  </div>,
]
`;

exports[`Prompt Component should match snapshot with blurred prop 1`] = `
Array [
  <div
    className="overlay"
    onClick={null}
    style={
      Object {
        "cursor": "inherit",
        "opacity": 0.5,
      }
    }
  />,
  <div
    className="prompt"
  >
    <div
      style={
        Object {
          "animation": "bcCCNc 300ms ease 0s 1 normal none running",
          "backfaceVisibility": "visible",
        }
      }
    >
      <div
        className="prompt-top"
      >
        <span
          className="prompt-title"
        >
          Title of the prompt component
        </span>
        <span
          onClick={[Function]}
        >
          <i
            className="material-icons"
          >
            close
          </i>
        </span>
      </div>
      <div
        className="prompt-middle"
        style={
          Object {
            "marginBottom": 9,
          }
        }
      >
        <span
          className="prompt-content"
        >
          Some content here..
        </span>
      </div>
      <div
        className="prompt-bottom"
      >
        <a
          className="sec_btn"
          href="#"
          onClick={[Function]}
        >
          Cancel
        </a>
        <a
          className="pri_btn prompt-done"
          href="#"
          onClick={[MockFunction]}
        >
          Delete
        </a>
      </div>
    </div>
  </div>,
]
`;
